tests/test_agrregation_region_aggrfuncs.py

```diff
@@ -0,0 +1,241 @@
+    np.testing.assert_allclose(
+        list(df1.filter(region='R1_world').data.value), [113.1, 142.4, 172.9])
+
+
+def test_df2_two_stage_sum(caplog):
+    """
+        sum in two stages
+        All := sum(north) + sum (othet)
+    """
+    r2_northregs = ['EEU', 'FSU']
+    r2_otherregs = ['AFR', 'CPA', 'LAM']
+
+    df1 = pyam.IamDataFrame(TEST2_DF)
+    df1.aggregate_region('FE', region='A1_All',
+                         subregions=all_regs, append=True)
+    df2 = pyam.IamDataFrame(TEST2_DF)
+    df2.aggregate_region('FE', region='A2_North', method='sum',
+                         subregions=r2_northregs, append=True)
+    df2.aggregate_region('FE', region='A2_Other', method='sum',
+                         subregions=r2_otherregs, append=True)
+    df2.aggregate_region('FE', region='A1_All', method='sum',
+                         subregions=['A2_North', 'A2_Other'], append=True)
+    pd.testing.assert_frame_equal(df1.filter(region='A1_All').data,
+                                  df2.filter(region='A1_All').data)
+    np.testing.assert_allclose(
+        list(df1.filter(region='A1_All').data.value), [113.1, 142.4, 172.9])
+
+
+def test_df2_avg(caplog):
+    df1 = pyam.IamDataFrame(TEST2_DF)
+    df1.aggregate_region('FE', region='R1_world', method=np.mean,
+                         subregions=all_regs, append=True)
+    df2 = pyam.IamDataFrame(TEST2_DF)
+    df2.aggregate_region('FE', region='R1_world', method='avg',
+                         subregions=all_regs, append=True)
+    pd.testing.assert_frame_equal(df1.data, df2.data)
+    np.testing.assert_allclose(
+        list(df1.filter(region='R1_world').data.value), [22.62, 28.48, 34.58])
+
+
+def test_df2_single_region_w_agg_with_None(caplog):
+    df = pyam.IamDataFrame(TEST2_DF)
+    df.data.loc[df.data['year'] == 2005, 'value'] = None
+    df.weighted_average_region('FE', region='A1_CPA',
+                               subregions=['CPA'],
+                               append=True, weight='POP')
+    exp_l = [v for v in list(df.filter(region='CPA',
+             variable='FE').data.value) if not math.isnan(v)]
+    obj_l = [v for v in list(df.filter(region='A1_CPA').data.value)
+             if not math.isnan(v)]
+    np.testing.assert_allclose(obj_l, exp_l)
+
+
+def test_df2_single_region_w_agg_with_var_zero(caplog):
+    df = pyam.IamDataFrame(TEST2_DF)
+    df.data.loc[df.data['variable'] == 'FE', 'value'] = 0.  # None
+    df.weighted_average_region('FE', region='A1_CPA',
+                               subregions=['CPA'],
+                               append=True, weight='POP')
+    exp_l = [v for v in list(df.filter(region='CPA',
+             variable='FE').data.value) if not math.isnan(v)]
+    obj_l = [v for v in list(df.filter(region='A1_CPA').data.value)
+             if not math.isnan(v)]
+    np.testing.assert_allclose(obj_l, exp_l)
+
+
+# @pytest.mark.skip(reason="w.avg does not work with zero or None values")
```

Comment:
There is some commented code in this file -- I'd either add a TODO with a short explanation if you plan to still update/use it, or remove the code otherwise